pkg/cache/scheduler_callback.go

```diff
@@ -56,11 +56,20 @@ func (callback *AsyncRMCallback) UpdateAllocation(response *si.AllocationRespons
 			zap.String("nodeID", alloc.NodeID))
 
 		// update cache
+		task := callback.context.getTask(alloc.ApplicationID, alloc.AllocationKey)
+		if task != nil {
+			task.setAllocationID(alloc.AllocationID)
+		} else {
+			log.Log(log.ShimRMCallback).Warn("Unable to get task", zap.String("taskID", alloc.AllocationKey))
+		}
 		if err := callback.context.AssumePod(alloc.AllocationKey, alloc.NodeID); err != nil {
+			if task != nil {
+				task.failWithEvent(err.Error(), "AssumePodError")
+			}
 			return err
 		}
 		if app := callback.context.GetApplication(alloc.ApplicationID); app != nil {
-			if task := callback.context.getTask(app.GetApplicationID(), alloc.AllocationKey); task != nil {
+			if task != nil {
```

Comment:
This needs a cleanup: `getTask()` already checks for the application. On top of that we have the appID already in the task so we do not need to get it from the app. That needs to be applied to the whole function (new jira)